crates/bevy_ecs/src/query/state.rs

```diff
@@ -2232,6 +2232,10 @@ mod tests {
         let mut query = QueryState::<Has<C>>::new(&mut world);
         assert_eq!(3, query.iter(&world).count());
 
+        // Allows should bypass the filter entirely
```

Comment:
Is it possible to write or update a test or example explaining why allows is different than has?